File: src/my_cpp_pkg/src/minimal_server.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/srv/add_two_ints.hpp"

/**
 * Server example that adds two integers.
 */
class MinimalServer : public rclcpp::Node
{
    public:

    /**
     * Constructor
     */
    MinimalServer() : Node("minimal_server")
    {
        // Create a service (server) object

        server_ = this->create_service<example_interfaces::srv::AddTwoInts>(
            "add_ints",
            std::bind(
                &MinimalServer::server_callback,
                this,
                std::placeholders::_1,
                std::placeholders::_2));
    }

    private:

    /**
     * Responds with sum of request integers
     */
     void server_callback (
         const example_interfaces::srv::AddTwoInts::Request::SharedPtr request,
         const example_interfaces::srv::AddTwoInts::Response::SharedPtr response)
     {
         response->sum = request->a + request->b;

         RCLCPP_INFO(this->get_logger(), "Incoming request\na: %ld\nb: %ld", request->a, request->b);
     }

    // Declare member variables
    rclcpp::Service<example_interfaces::srv::AddTwoInts>::SharedPtr server_;
};

/**
 * Main entrypoint
 */
int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<MinimalServer>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: src/my_py_pkg/my_py_pkg/minimal_server.py
import rclpy
from example_interfaces.srv import AddTwoInts
from rclpy.executors import ExternalShutdownException
from rclpy.node import Node


class MinimalService(Node):
    """Minimal service node that adds two integers."""

    def __init__(self):
        # Call the Node class constructor with the node name
        super().__init__("minimal_server")

        # Create a service object
        self.srv = self.create_service(
            AddTwoInts,
            "add_ints",
            self._server_callback,
        )

    def _server_callback(self, req, resp):
        """Respond to a request with sum of two integers"""
        resp.sum = req.a + req.b
        self.get_logger().info("Incoming request\na: %d b: %d" % (req.a, req.b))
        return resp


def main():
    rclpy.init()
    node = MinimalService()
    try:
        rclpy.spin(node)
    except (KeyboardInterrupt, ExternalShutdownException):
        pass
    finally:
        node.destroy_node()
        rclpy.shutdown()


if __name__ == "__main__":
    main()
